lib/bundler/shared_helpers.rb

```diff
@@ -245,7 +255,12 @@ def validate_bundle_path
     end
 
     def find_gemfile(order_matters = false)
-      given = ENV["BUNDLE_GEMFILE"]
+      require "bundler"
+      given = if Bundler.feature_flag.config_relative_to_cwd?
+        Bundler.settings[:gemfile]
+      else
+        ENV["BUNDLE_GEMFILE"]
+      end
       return given if given && !given.empty?
       names = gemfile_names
       names.reverse! if order_matters && Bundler.feature_flag.prefer_gems_rb?
```

Comment:
As a side note, it took me a while to figure out that I needed to `require "bundler"` in this method, because I assumed the presence of `Bundler.feature_flag.prefer_gems_rb?` here implied that it was already available at this point. Turns out that this condition is never executed by our specs (or at least removing it does not cause any failures). So, maybe we need some coverage there, and also maybe the addition of `require "bundler"` is fixing some issue with the `prefer_gems_rb` feature flag... :man_shrugging:.